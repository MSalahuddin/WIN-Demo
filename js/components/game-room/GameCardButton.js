import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import {Platform} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import Text, { FONT_FAMILY, SIZE } from '../common/Text';
import Button, { BUTTON_COLOR_SCHEME } from '../common/Button';
import { scale, scaleHeight, heightRatio } from '../../platformUtils';
import { color } from '../../styles';
import { coin } from '../../../assets/images';
import { gameRoomStrings } from '../../stringConstants';
import { convertNumberToStringWithComma } from '../../utils';

const OverlayButtonContainer = styled.View`
  position: absolute;
  right: ${scale(-5)};
  top: ${({ top }) => scaleHeight(top)};
`;

const ButtonContentContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: center;
`;

const TextContainer = styled.View`
  justify-content: center;
`;

const ButtonText = styled(Text)`
  margin-top: ${({ marginTop }) => (heightRatio > 1 ? scaleHeight(marginTop) : scaleHeight(marginTop / heightRatio))};
  margin-horizontal: ${scale(5)};
`;

const CoinIcon = styled.Image`
  margin-horizontal: ${({ marginHorizontal }) => scale(marginHorizontal)};
  height: ${({ size }) => scale(size)};
  width: ${({ size }) => scale(size)};
`;

const RedStrikeLine = styled.View`
  border-bottom-width: ${scale(2)};
  border-color: ${color.pinkRed};
  height: 1;
  position: absolute;
  width: ${({ width }) => `${width}`};
`;
const PlayButton = styled.View`
width :60%;
align-self:flex-end;

`;
const PlayButtonContainer = styled(LinearGradient)`
border-bottom-right-radius:10;
padding-vertical:${scale(5)};
height:${scale(39)}
`;

const GameCardButton = ({ isDisabled, isLarge, isFree, isMachine, isGameVIP, tokensCost, onPress, playLabel,
  gameCard,
  isGradient, 
  onLayout, ...rest }) => {
  const buttonHeight = isLarge ? 80 : 48;
  const buttonWidth = isLarge ? 271 : 140;
  const buttonTextSize = isLarge ? SIZE.XLARGE : SIZE.SMALL;
  const iconSize = isLarge ? 40 : 29;
  const iconMarginHorizontal = isLarge ? 8 : 5;

  const renderOverlayButton = () => {
    let overlayButtonHeight = 25;
    let overlayButtonWidth = isFree ? 65 : 140;
    let borderRadius = isFree ? 40 : 24;
    let marginTop =Platform.OS === "android"? -3 : 2;
    let top = -25;
    const buttonText = isMachine ? gameRoomStrings.play : isFree ? gameRoomStrings.free : gameRoomStrings.becomeVip;
    const theme = isFree ? BUTTON_COLOR_SCHEME.RED : BUTTON_COLOR_SCHEME.PURPLE;

    if (isLarge) {
      borderRadius = isFree ? 53 : 83;
      overlayButtonWidth = isFree ? 74 : 132;
      overlayButtonHeight = 35;
      marginTop =Platform.OS === "android"? 0 : 5;
      top = -28;
    }

    return (
      <OverlayButtonContainer top={top}>
        <Button
          testID="overlay-button"
          borderRadius={borderRadius}
          height={overlayButtonHeight}
          width={overlayButtonWidth}
          theme={theme}
          onPress={onPress}
        >
          <ButtonText color={color.white} fontFamily={FONT_FAMILY.BOLD_ITALIC} size={SIZE.SMALL} marginTop={marginTop}>
            {buttonText}
          </ButtonText>
        </Button>
      </OverlayButtonContainer>
    );
  };

  return (
    <>
   {!gameCard && <Button
    testID="play-button"
    borderRadis={53}
    height={buttonHeight}
    width={buttonWidth}
    gardientHighlight={isGradient}
                
    gradientColor={[color.gradientLightBlue,color.gradientDarkBlue,color.bottomNavColor3]}
    theme={isGradient?BUTTON_COLOR_SCHEME.WHITE:BUTTON_COLOR_SCHEME.GREEN}
    onPress={onPress}
    onLayout={onLayout}
    {...rest}
  >
    <ButtonContentContainer>
      <ButtonText size={buttonTextSize} color={color.white} marginTop={Platform.OS==="android"? 0 : 5}>
        {playLabel}
      </ButtonText>
      <CoinIcon source={coin} size={iconSize} marginHorizontal={iconMarginHorizontal} resizeMode="contain" />
      <TextContainer>
        <ButtonText 
          size={buttonTextSize} 
          color={color.white} 
          fontFamily={FONT_FAMILY.BOLD} 
          marginTop={Platform.OS==="android"? 0 : 5}
        >
          {convertNumberToStringWithComma(tokensCost)}
        </ButtonText>
        {isDisabled && <RedStrikeLine width="85%" />}
      </TextContainer>
      {isFree && <RedStrikeLine width={isLarge ? '65%' : '85%'} />}
    </ButtonContentContainer>
    {(isDisabled || isFree) && renderOverlayButton()}
  </Button>
  }
    {gameCard && <PlayButton
      {...rest}
    >
      <PlayButtonContainer
       start={{ x: 0, y: 0 }} end={{ x: 1, y: 0 }}
         colors={[color.gameCardGradient1,color.gameCardGradient2]}
      >
      <ButtonContentContainer>
        {/* <ButtonText size={buttonTextSize} color={color.white} marginTop={Platform.OS==="android"? 0 : 5}>
          {playLabel}
        </ButtonText> */}
         <TextContainer>
          <ButtonText 
            size={SIZE.LARGE} 
            color={color.white} 
            fontFamily={FONT_FAMILY.BOLD} 
            marginTop={Platform.OS==="android"? 0 : 5}
          >
            {isMachine ? gameRoomStrings.play?.toUpperCase() : isFree?gameRoomStrings.free:convertNumberToStringWithComma(tokensCost)}
            
          </ButtonText>
          </TextContainer>
        {!isFree && <CoinIcon source={coin} size={25} marginHorizontal={iconMarginHorizontal} resizeMode="contain" />}
     
      </ButtonContentContainer>
      </PlayButtonContainer>
    </PlayButton>}
    </>
  );
};

GameCardButton.propTypes = {
  onLayout:PropTypes.func,
  isLarge: PropTypes.bool,
  isFree: PropTypes.bool,
  isMachine: PropTypes.bool,
  isGameVIP: PropTypes.bool,
  isDisabled: PropTypes.bool,
  tokensCost: PropTypes.number.isRequired,
  onPress: PropTypes.func.isRequired,
  playLabel: PropTypes.string,
  gameCard:PropTypes.bool,
  isGradient:PropTypes.bool
};

GameCardButton.defaultProps = {
  isLarge: false,
  isFree: false,
  isMachine: false,
  isGameVIP: false,
  isDisabled: false,
  playLabel: gameRoomStrings.play,
  onLayout: () => {} ,
  gameCard:false,
  isGradient:false
};

export default GameCardButton;
