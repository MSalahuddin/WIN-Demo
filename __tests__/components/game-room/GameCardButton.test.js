import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import GameCardButton from '../../../js/components/game-room/GameCardButton';
import MockProvider from '../../MockProvider';

describe('Component GameCardButton', () => {
  const onPress = jest.fn();

  test('renders regular button correctly', () => {
    const output = render(
      <MockProvider>
        <GameCardButton tokensCost={100} onPress={onPress} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders free button correctly', () => {
    const output = render(
      <MockProvider>
        <GameCardButton tokensCost={100} isFree onPress={onPress} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders become VIP button for non-VIP user correctly', () => {
    const output = render(
      <MockProvider>
        <GameCardButton tokensCost={100} isGameVIP onPress={onPress} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly isDisabled', () => {
    const output = render(
      <MockProvider>
        <GameCardButton tokensCost={100} isDisabled onPress={onPress} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when play button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <GameCardButton tokensCost={100} onPress={onPress} />
      </MockProvider>
    );
    fireEvent(getByTestId('play-button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });

  test('calls onPress when free button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <GameCardButton tokensCost={100} onPress={onPress} isFree />
      </MockProvider>
    );
    fireEvent(getByTestId('overlay-button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });
});
