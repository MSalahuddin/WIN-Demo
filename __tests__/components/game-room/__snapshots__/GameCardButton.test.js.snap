// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component GameCardButton renders become VIP button for non-VIP user correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(96,202,142,1)",
        "borderRadius": 80,
        "height": 96,
        "justifyContent": "center",
        "opacity": 1,
        "width": 280,
      }
    }
    testID="play-button"
  >
    <View
      borderRadius={40}
      colorScheme="rgba(128,229,173,1)"
      height={48}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(128,229,173,1)",
            "borderRadius": 80,
            "height": 86,
            "justifyContent": "center",
            "width": 274,
          },
        ]
      }
      width={140}
    >
      <View
        borderRadius={40}
        colorScheme="rgba(114,213,158,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(114,213,158,1)",
              "borderRadius": 100,
              "height": 82,
              "justifyContent": "center",
              "marginTop": 8,
              "width": 274,
            },
          ]
        }
        width={140}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            marginTop={5}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 7.5,
                  "marginTop": 10,
                },
              ]
            }
          >
            Play
          </Text>
          <Image
            marginHorizontal={5}
            resizeMode="contain"
            size={29}
            source={1}
            style={
              Array [
                Object {
                  "height": 43.5,
                  "marginHorizontal": 7.5,
                  "width": 43.5,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              alignCenter={false}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              marginTop={5}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "left",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginHorizontal": 7.5,
                    "marginTop": 10,
                  },
                ]
              }
            >
              100
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component GameCardButton renders correctly isDisabled 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(96,202,142,1)",
        "borderRadius": 80,
        "height": 96,
        "justifyContent": "center",
        "opacity": 1,
        "width": 280,
      }
    }
    testID="play-button"
  >
    <View
      borderRadius={40}
      colorScheme="rgba(128,229,173,1)"
      height={48}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(128,229,173,1)",
            "borderRadius": 80,
            "height": 86,
            "justifyContent": "center",
            "width": 274,
          },
        ]
      }
      width={140}
    >
      <View
        borderRadius={40}
        colorScheme="rgba(114,213,158,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(114,213,158,1)",
              "borderRadius": 100,
              "height": 82,
              "justifyContent": "center",
              "marginTop": 8,
              "width": 274,
            },
          ]
        }
        width={140}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            marginTop={5}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 7.5,
                  "marginTop": 10,
                },
              ]
            }
          >
            Play
          </Text>
          <Image
            marginHorizontal={5}
            resizeMode="contain"
            size={29}
            source={1}
            style={
              Array [
                Object {
                  "height": 43.5,
                  "marginHorizontal": 7.5,
                  "width": 43.5,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              alignCenter={false}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              marginTop={5}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "left",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginHorizontal": 7.5,
                    "marginTop": 10,
                  },
                ]
              }
            >
              100
            </Text>
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 3,
                    "borderColor": "#F06B59",
                    "height": 1,
                    "position": "absolute",
                    "width": "85%",
                  },
                ]
              }
              width="85%"
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "position": "absolute",
                "right": -7.5,
                "top": -50,
              },
            ]
          }
          top={-25}
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "marginVertical": 10,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(95,110,175,1)",
                  "borderRadius": 48,
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 280,
                }
              }
              testID="overlay-button"
            >
              <View
                borderRadius={24}
                colorScheme="rgba(140,158,221,1)"
                height={25}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(140,158,221,1)",
                      "borderRadius": 48,
                      "height": 40,
                      "justifyContent": "center",
                      "width": 274,
                    },
                  ]
                }
                width={140}
              >
                <View
                  borderRadius={24}
                  colorScheme="rgba(124,137,216,1)"
                  height={25}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(124,137,216,1)",
                        "borderRadius": 68,
                        "height": 36,
                        "justifyContent": "center",
                        "marginTop": 8,
                        "width": 274,
                      },
                    ]
                  }
                  width={140}
                >
                  <Text
                    alignCenter={false}
                    allowFontScaling={true}
                    color="#FFFFFF"
                    fontFamily="BOLD ITALIC"
                    isUnderlined={false}
                    isUppercase={false}
                    marginTop={2}
                    maxFontSizeMultiplier={1.3}
                    size="SMALL"
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontFamily": "CalibreBoldItalic",
                          "fontSize": 27,
                          "justifyContent": "center",
                          "textAlign": "left",
                          "textDecorationColor": "#FFFFFF",
                          "textDecorationLine": "none",
                          "textDecorationStyle": "solid",
                          "textTransform": "none",
                        },
                        Object {
                          "marginHorizontal": 7.5,
                          "marginTop": 4,
                        },
                      ]
                    }
                  >
                    Become a VIP
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component GameCardButton renders free button correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(96,202,142,1)",
        "borderRadius": 80,
        "height": 96,
        "justifyContent": "center",
        "opacity": 1,
        "width": 280,
      }
    }
    testID="play-button"
  >
    <View
      borderRadius={40}
      colorScheme="rgba(128,229,173,1)"
      height={48}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(128,229,173,1)",
            "borderRadius": 80,
            "height": 86,
            "justifyContent": "center",
            "width": 274,
          },
        ]
      }
      width={140}
    >
      <View
        borderRadius={40}
        colorScheme="rgba(114,213,158,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(114,213,158,1)",
              "borderRadius": 100,
              "height": 82,
              "justifyContent": "center",
              "marginTop": 8,
              "width": 274,
            },
          ]
        }
        width={140}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            marginTop={5}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 7.5,
                  "marginTop": 10,
                },
              ]
            }
          >
            Play
          </Text>
          <Image
            marginHorizontal={5}
            resizeMode="contain"
            size={29}
            source={1}
            style={
              Array [
                Object {
                  "height": 43.5,
                  "marginHorizontal": 7.5,
                  "width": 43.5,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              alignCenter={false}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              marginTop={5}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "left",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginHorizontal": 7.5,
                    "marginTop": 10,
                  },
                ]
              }
            >
              100
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomWidth": 3,
                  "borderColor": "#F06B59",
                  "height": 1,
                  "position": "absolute",
                  "width": "85%",
                },
              ]
            }
            width="85%"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "position": "absolute",
                "right": -7.5,
                "top": -50,
              },
            ]
          }
          top={-25}
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 10,
                  "marginVertical": 10,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(197,78,62,1)",
                  "borderRadius": 80,
                  "height": 50,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 130,
                }
              }
              testID="overlay-button"
            >
              <View
                borderRadius={40}
                colorScheme="rgba(255,132,115,1)"
                height={25}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(255,132,115,1)",
                      "borderRadius": 80,
                      "height": 40,
                      "justifyContent": "center",
                      "width": 124,
                    },
                  ]
                }
                width={65}
              >
                <View
                  borderRadius={40}
                  colorScheme="rgba(240, 107, 89, 1)"
                  height={25}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgba(240, 107, 89, 1)",
                        "borderRadius": 100,
                        "height": 36,
                        "justifyContent": "center",
                        "marginTop": 8,
                        "width": 124,
                      },
                    ]
                  }
                  width={65}
                >
                  <Text
                    alignCenter={false}
                    allowFontScaling={true}
                    color="#FFFFFF"
                    fontFamily="BOLD ITALIC"
                    isUnderlined={false}
                    isUppercase={false}
                    marginTop={2}
                    maxFontSizeMultiplier={1.3}
                    size="SMALL"
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontFamily": "CalibreBoldItalic",
                          "fontSize": 27,
                          "justifyContent": "center",
                          "textAlign": "left",
                          "textDecorationColor": "#FFFFFF",
                          "textDecorationLine": "none",
                          "textDecorationStyle": "solid",
                          "textTransform": "none",
                        },
                        Object {
                          "marginHorizontal": 7.5,
                          "marginTop": 4,
                        },
                      ]
                    }
                  >
                    FREE!
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Component GameCardButton renders regular button correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginHorizontal": 10,
        "marginVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(96,202,142,1)",
        "borderRadius": 80,
        "height": 96,
        "justifyContent": "center",
        "opacity": 1,
        "width": 280,
      }
    }
    testID="play-button"
  >
    <View
      borderRadius={40}
      colorScheme="rgba(128,229,173,1)"
      height={48}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(128,229,173,1)",
            "borderRadius": 80,
            "height": 86,
            "justifyContent": "center",
            "width": 274,
          },
        ]
      }
      width={140}
    >
      <View
        borderRadius={40}
        colorScheme="rgba(114,213,158,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(114,213,158,1)",
              "borderRadius": 100,
              "height": 82,
              "justifyContent": "center",
              "marginTop": 8,
              "width": 274,
            },
          ]
        }
        width={140}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            alignCenter={false}
            allowFontScaling={true}
            color="#FFFFFF"
            fontFamily="MEDIUM"
            isUnderlined={false}
            isUppercase={false}
            marginTop={5}
            maxFontSizeMultiplier={1.3}
            size="SMALL"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Calibre-Medium",
                  "fontSize": 27,
                  "justifyContent": "center",
                  "textAlign": "left",
                  "textDecorationColor": "#FFFFFF",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                  "textTransform": "none",
                },
                Object {
                  "marginHorizontal": 7.5,
                  "marginTop": 10,
                },
              ]
            }
          >
            Play
          </Text>
          <Image
            marginHorizontal={5}
            resizeMode="contain"
            size={29}
            source={1}
            style={
              Array [
                Object {
                  "height": 43.5,
                  "marginHorizontal": 7.5,
                  "width": 43.5,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              alignCenter={false}
              allowFontScaling={true}
              color="#FFFFFF"
              fontFamily="BOLD"
              isUnderlined={false}
              isUppercase={false}
              marginTop={5}
              maxFontSizeMultiplier={1.3}
              size="SMALL"
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "CalibreBold",
                    "fontSize": 27,
                    "justifyContent": "center",
                    "textAlign": "left",
                    "textDecorationColor": "#FFFFFF",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                    "textTransform": "none",
                  },
                  Object {
                    "marginHorizontal": 7.5,
                    "marginTop": 10,
                  },
                ]
              }
            >
              100
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
